src/main/java/uk/gov/hmcts/reform/cwrdapi/util/AuditInterceptor.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.hmcts.reform.cwrdapi.util;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.multipart.MultipartHttpServletRequest;
+import org.springframework.web.servlet.HandlerInterceptor;
+import uk.gov.hmcts.reform.cwrdapi.controllers.advice.ForbiddenException;
+import uk.gov.hmcts.reform.cwrdapi.oidc.JwtGrantedAuthoritiesConverter;
+import uk.gov.hmcts.reform.cwrdapi.repository.AuditRepository;
+import uk.gov.hmcts.reform.cwrdapi.service.IValidationService;
+
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.constraints.NotNull;
+
+import static java.util.Objects.nonNull;
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static uk.gov.hmcts.reform.cwrdapi.controllers.constants.ErrorConstants.FILE_UPLOAD_IN_PROGRESS;
+import static uk.gov.hmcts.reform.cwrdapi.util.AuditStatus.IN_PROGRESS;
+import static uk.gov.hmcts.reform.cwrdapi.util.CaseWorkerConstants.FILE;
+
+@MultipartConfig
+@Component
+public class AuditInterceptor implements HandlerInterceptor {
+
+    @Autowired
+    IValidationService validationServiceFacadeImpl;
+
+    @Autowired
+    AuditRepository auditRepository;
+
+    @Autowired
+    @Lazy
+    private JwtGrantedAuthoritiesConverter jwtGrantedAuthoritiesConverter;
+
+    @Override
+    public boolean preHandle(HttpServletRequest request,
+                             @NotNull HttpServletResponse response, @NotNull Object handler) {
```

Comment:
Will this intercepter be called after SIDAM filters (s2s and bearer token validation are done)are called or before that ? nothing to worry if this is being called after s2s and bearer token is done,

Reply:
yes it is called after that.